pkg/segment/query/processor/statscommand.go

```diff
@@ -18,22 +18,323 @@
+		if err != nil {
+			return nil, toputils.TeeErrorf("stats.Process: cannot initialize search results; err=%v", err)
+		}
+		p.searchResults = searchResults
+	}
+
+	blkResults := p.searchResults.BlockResults
+
+	measureInfo, internalMops := blkResults.GetConvertedMeasureInfo()
+	measureResults := make([]utils.CValueEnclosure, len(internalMops))
+
+	for i := 0; i < numOfRecords; i++ {
+		record := inputIQR.GetRecord(i)
+
+		// Bucket Key
+		bucketKeyBufIdx := 0
+
+		for idx, cname := range p.options.GroupByRequest.GroupByColumns {
+			if idx > 0 {
+				copy(p.bucketKeyWorkingBuf[bucketKeyBufIdx:], utils.BYTE_TILDE)
+				bucketKeyBufIdx += utils.BYTE_TILDE_LEN
+			}
+
+			cValue, err := record.ReadColumn(cname)
+			if err != nil {
+				p.errorData.readColumns[cname] = struct{}{}
+				copy(p.bucketKeyWorkingBuf[bucketKeyBufIdx:], utils.VALTYPE_ENC_BACKFILL)
+				bucketKeyBufIdx += 1
+			} else {
+				bytesVal := cValue.ConvertToBytesValue()
+				copy(p.bucketKeyWorkingBuf[bucketKeyBufIdx:], bytesVal)
+				bucketKeyBufIdx += len(bytesVal)
+			}
+		}
+
+		for cname, indices := range measureInfo {
+			cValue, err := record.ReadColumn(cname)
+			if err != nil {
+				p.errorData.readColumns[cname] = struct{}{}
+				cValue = &utils.CValueEnclosure{CVal: utils.VALTYPE_ENC_BACKFILL, Dtype: utils.SS_DT_BACKFILL}
+			}
+
+			for _, idx := range indices {
+				measureResults[idx] = *cValue
+			}
+		}
+		blkResults.AddMeasureResultsToKey(p.bucketKeyWorkingBuf[:bucketKeyBufIdx], measureResults, "", false, inputIQR.GetQID())
+	}
+
+	return nil, nil
+}
+
+func (p *statsProcessor) ExtractGroupByResults(iqr *iqr.IQR) (*iqr.IQR, error) {
+	if p.searchResults == nil {
+		return iqr, nil
+	}
+
+	// load and convert the bucket results
+	_ = p.searchResults.GetBucketResults()
+
+	bucketHolderArr, retMFuns, aggGroupByCols, _, _ := p.searchResults.GetGroupyByBuckets(int(utils.QUERY_MAX_BUCKETS))
+
+	knownValues := make(map[string][]utils.CValueEnclosure)
+
+	for _, aggGroupByCol := range aggGroupByCols {
+		knownValues[aggGroupByCol] = make([]utils.CValueEnclosure, len(bucketHolderArr))
+	}
+	for _, retMFun := range retMFuns {
+		knownValues[retMFun] = make([]utils.CValueEnclosure, len(bucketHolderArr))
+	}
+
+	for i, bucketHolder := range bucketHolderArr {
+		for idx, aggGroupByCol := range aggGroupByCols {
+			colValue := bucketHolder.GroupByValues[idx]
+			err := knownValues[aggGroupByCol][i].ConvertValue(colValue)
+			if err != nil {
+				p.errorData.convertToCValueErr[fmt.Sprintf("%v_%v", i, aggGroupByCol)] = colValue
+			}
+		}
+
+		for _, retMFun := range retMFuns {
+			value := bucketHolder.MeasureVal[retMFun]
+			err := knownValues[retMFun][i].ConvertValue(value)
+			if err != nil {
+				p.errorData.convertToCValueErr[fmt.Sprintf("%v_%v", i, retMFun)] = value
+			}
+		}
+	}
+
+	err := iqr.AppendKnownValues(knownValues)
+	if err != nil {
+		return nil, toputils.TeeErrorf("stats.Process: cannot append known values; err=%v", err)
+	}
+
+	p.logErrors()
```

Comment:
make sure we do not log all the error log entries, (see comment in other PR). just log the first 5 entries.

Reply:
The errors that would be logged here are only for the columns, that are given as `groupByColumns` and the `measureColumns`